src/AiClient.php

```diff
@@ -114,14 +115,44 @@ public static function defaultRegistry(): ProviderRegistry
     /**
      * Checks if a provider is configured and available for use.
      *
+     * Supports multiple input formats for developer convenience:
+     * - ProviderAvailabilityInterface: Direct availability check (backward compatible)
```

Comment:
This isn't purely for backward compatibility, we just want to support all 3 ways.
```suggestion
     * - ProviderAvailabilityInterface: Direct availability check
```